ipv6: Support Tokenised IPv6 Identifiers

cathay4t opened the pull request:
Similar to `ip token set ::123/64 dev eth1` which setting the IPv6
address retried from IPv6 RA/autoconf to `<ra_prefix>::<token>`.

Example yaml:

```yml
---
interfaces:
- name: eth1
  state: up
  ipv6:
    token: ::fac1
    autoconf: true
    dhcp: true
    enabled: true
```

To remove the IPv6 token, you may set use:
 * Empty string: `token: ""`
 * Explicitly all zero: `token: "::"`

Due to kernel limitation, you cannot set IPv6 token to all zero.

You will get invalid argument error when:
 * Specified token not been started with 64 bits of 0.
 * Setting `autoconf: false` with non-empty token and non-all-zero token.

Integration test cases included.

ffmancera commented on rust/src/lib/ip.rs:
```diff
@@ -558,6 +565,20 @@ impl InterfaceIpv6 {
+                return Err(NmstateError::new(
+                    ErrorKind::InvalidArgument,
+                    format!(
+                        "Desired IPv6 token '{token}' cannot \
+                        be applied with IPv6 autoconf disabled"
```
Just a nit, should we suggest the user to configure the IPv6 token to "" or "::"? What do you think?

ffmancera commented on rust/src/lib/ip.rs:
```diff
@@ -1143,3 +1171,51 @@ impl MergedInterface {
+) -> Result<(), NmstateError> {
+    // Empty token means reverting to default "::"
+    if token.is_empty() {
+        token.clear();
+        write!(token, "::").ok();
```
If the string is empty it means it has a length of zero. And at the same time `clear()`  will make the string have a length of zero (empty). So I think checking if it is empty and then clearing is overkill IMO. What do you think?

ffmancera commented on rust/src/lib/ip.rs:
```diff
@@ -1143,3 +1171,51 @@ impl MergedInterface {
+                // The Ipv6Addr::to_string() will convert
+                //  ::fac1 to ::0.0.250.193
+                // Which is no ideal in this case
+                // To workaround that, we set leading 64 bites to '2001:db8::',
```
I think there is a typo in the comment. Did you mean "bits" instead of "bites"?

ffmancera reviewed (changes requested):
In general it looks good, there is just a couple of nits. Thank you for the work!

cathay4t committed:
ipv6: Support Tokenised IPv6 Identifiers

Similar to `ip token set ::123/64 dev eth1` which setting the IPv6
address retried from IPv6 RA/autoconf to `<ra_prefix>::<token>`.

Example yaml:

```yml
---
interfaces:
- name: eth1
  state: up
  ipv6:
    token: ::fac1
    autoconf: true
    dhcp: true
    enabled: true
```

To remove the IPv6 token, you may set use:
 * Empty string: `token: ""`
 * Explicitly all zero: `token: "::"`

Due to kernel limitation, you cannot set IPv6 token to all zero.

You will get invalid argument error when:
 * Specified token not been started with 64 bits of 0.
 * Setting `autoconf: false` with non-empty token and non-all-zero token.

Integration test cases included.

Signed-off-by: Gris Ge <fge@redhat.com>
